packages/@azure/servicebus/data-plane/lib/session/messageSession.ts

```diff
@@ -590,6 +593,10 @@ export class MessageSession extends LinkEntity {
             }
           }
           return;
+        } finally {
+          if (this._receiver) {
+            this._receiver!.addCredit(1);
```

Comment:
I thought we werent doing the credit related changes in this PR i.e. this PR is only for surfacing the error when settling messages or other actions.

Can we have this in a separate PR?

Reply:
Ah! Never mind, I saw your comment https://github.com/Azure/azure-sdk-for-js/issues/1051#issuecomment-461264529 only now.